lib/autoproj/configuration.rb

```diff
@@ -606,5 +606,34 @@ def to_hash
             end
             result
         end
+
+        # Allows to load a seed-config.yml file (also from a buildconf repository)
+        # rather than providing it before checkout using the --seed-config paramater
+        # of the autoproj_bootstrap script
+        # this allows to bootstrap with --no-interactive and still apply a custom config e.g. in CI/CD
+        # The call to this function has to be in the init.rb of the buildconf BEFORE any other
+        # config option, e.g. the git server configuration settings
+        # The filename parameter is the name of the config seed yml file in the repository
+        def load_config_once(filename)
+            unless Autoproj.config.has_value_for?("default_config_applied_#{filename}")
```

Comment:
Do an early return instead of a gigantic if.

~~~
return if get("default_config_applied_#{filename}", false)

<rest of method>
~~~